src/notifications/notificationpreviewpresenter.cpp

```diff
@@ -203,9 +224,10 @@ void NotificationPreviewPresenter::setCurrentNotification(LipstickNotification *
 {
     if (currentNotification != notification) {
         if (currentNotification) {
-            const bool displayWasOn(currentNotification->hints().value(NotificationManager::HINT_DISPLAY_ON).toBool());
-            if (displayWasOn) {
-                // Release our screen wake
+            const bool notificationWasCritical = currentNotification->urgency() >= 2 ||
+                                                 currentNotification->hints().value(NotificationManager::HINT_DISPLAY_ON).toBool();
+            if (notificationWasCritical) {
```

Comment:
Well, it should be symmetrical with the variable used for the new notification. We could use 'turnDisplayOn'/'displayWasTurnedOn', for example.  On the other hand, the criteria for enabling the display is the same logical test called 'notificationIsCritical' elsewhere in the code, so I think that is the clearest name.